### ISSUE TITLE
Allow providing annotations for Ingress resources created by the operator

### ISSUE THREAD

**Issue by ctron** [keycloak/keycloak#12592]
### Description

Right now it seems not possible to provide additional annotations for the ingress object, created by the operator, through the Keycloak resource.

### Motivation

I would expect that is it possible to provide custom annotations, as this is a common way with ingress to configure the routes/ingress handling.

Many of those extensions are vendor specific, it doesn't make sense to hard-code them all in Keycloak. The keycloak operator should still provide reasonable defaults. But it should be possible to override them.

**Comment by losblancos113** [comment]
I also need this feature. I'm using custom ingress but somehow it has been deleted (by operator??) after being created for a couple of hours.

**Comment by ombarek** [comment]
Hi, I've also the same request and the same issue. I add ingress.enable: false in the keycloak CRD and create my own ingress (specially for tls with let's encrypt and DNS challenge).
But my ingress has been deleted by the operator (last quarkus version use, 20.0.1) with a random delay. Operator seems delete the ingress without searching specific labels (ex: app: keycloak or managed by operator)

**Comment by rustygreen** [comment]
Any update on this?

**Comment by losblancos113** [comment]
> Hi, I've also the same request and the same issue. I add ingress.enable: false in the keycloak CRD and create my own ingress (specially for tls with let's encrypt and DNS challenge). But my ingress has been deleted by the operator (last quarkus version use, 20.0.1) with a random delay. Operator seems delete the ingress without searching specific labels (ex: app: keycloak or managed by operator)

You should use a different name for your custom ingress

**Comment by ombarek** [comment]
Yes, @losblancos113  I did it. I create another ingress with a different name and it resolve my problem.

**Comment by shawkins** [comment]
The keycloak cr support annotations under the ingress:

```
spec:
  ingress:
    annotations:
      custom: annotation
```
